### PR TITLE
Feature block retreat exchange option #rm5029

### PR DESCRIPTION
**Pull request by RomainFayolle**
| Q                                          | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                     | Enhancement
| Tickets (_issues_) concerned               | 5029

---

##### What have you changed ?
Add security for retreat exchange

##### How did you change it ?
Catch the 3 cases where exchanged is not allowed (price sup, price inf or retreat has options)

##### Is there a special consideration?
Ticket required to allow viewset to filter retreat by those without options. It's impossible at filter level because retreat.options is a property. However with Pedro we saw that a similar problem occurred before in with retreat.remaining_places and the problem was solved by filtering the field in the frontend cf image below. Should we fix the issue here with the same solution ?
image

Verification :
 -  [x] My branch name respect the standard defined in `CONTRIBUTING.md`
 -  [x] All my commits respect the standard defined in `CONTRIBUTING.md`
 -  [x] My additions are I18N
 -  [x] This Pull-Request fully meets the requirements defined in the issue
 -  [x] I added or modified the attached tests
 -  [x] I added or modified the documentation

### CODE REVIEW

File: retirement/serializers.py [hunk 1]
```diff
@@ -558,32 +566,44 @@ def update(self, instance, validated_data):
...
-                                "the current one. Provide a payment_token or "
-                                "single_use_token to charge the balance."
-                            )]
-                        })
+                    # need_transaction = True
+                    # amount = (
+                    #     validated_data['retreat'].price -
+                    #     order_line.coupon_real_value
+                    # )
+                    # if not (payment_token or single_use_token):
```
Comment (RignonNoel) [inline retirement/serializers.py]: No need to keep commented code without any explanation

File: retirement/serializers.py [hunk 2]
```diff
@@ -558,32 +566,44 @@ def update(self, instance, validated_data):
...
-                    price_diff = (
-                        current_retreat.price -
-                        validated_data['retreat'].price
-                    )
-                    real_cost = order_line.total_cost
-                    amount = min(price_diff, real_cost)
+                    # need_refund = True
+                    # price_diff = (
+                    #     current_retreat.price -
+                    #     validated_data['retreat'].price
```
Comment (RignonNoel) [inline retirement/serializers.py]: No need to keep commented code without any explanation

File: retirement/tests/tests_viewset_Reservation_update.py [hunk 3]
```diff
@@ -654,6 +655,7 @@ def test_update_partial_more_expensive_retreat_missing_info(self):
         self.retreat2.price = 199
         self.retreat2.save()
 
+    @skip
```
Comment (RignonNoel) [inline retirement/tests/tests_viewset_Reservation_update.py]: Any reason why we skip all these tests ? Why not removing them or updating them ?

**Review by RignonNoel — CHANGES_REQUESTED** [review]

File: retirement/serializers.py (hunk 1 above)
Comment (RomainFayolle) [reply retirement/serializers.py]: code removed and linked to git commit

File: retirement/serializers.py (hunk 2 above)
Comment (RomainFayolle) [reply retirement/serializers.py]: code removed and linked to git commit

File: retirement/tests/tests_viewset_Reservation_update.py (hunk 3 above)
Comment (RomainFayolle) [reply retirement/tests/tests_viewset_Reservation_update.py]: removed

**Review by RignonNoel — APPROVED** [review]

**Pull request merged**